app/lib/server/functions/addUserToRoom.js

```diff
@@ -27,6 +29,16 @@ export const addUserToRoom = function(rid, user, inviter, silenced) {
 		callbacks.run('beforeJoinRoom', user, room);
 	}
 
+	try {
```

Comment:
I think this should run before the callbacks, WDYT?

Reply:
My thinking to put it here was that "things in the core must run before the apps", but I don't really have a strong case about it. Why do you think it has to run before the callbacks?

Reply:
Because the callbacks may execute things thinking that the room will be created

Reply:
Solved [here](https://github.com/RocketChat/Rocket.Chat/pull/17656/files#diff-59771c3aa115644893a53436db956059)